### PR TITLE
XRDDEV-1893 Pinia

### PR DESCRIPTION
**Pull request by Riippi**
https://jira.niis.org/browse/XRDDEV-1893

- Change vuex to pinia in security server
- Remove vues and router from shared-ui because they were not used
- Fixes an probable issue in checking the session status. Previously vuex getter was called when it should have been an action. Meaning it didn't work.
- Clean up some unused code
- Fixes the initialisation wizard bug https://jira.niis.org/browse/XRDDEV-1890

### CODE REVIEW

**Review by teemukin65 — COMMENTED** [review]
Massive pull request, some comments, I learned something during the journey.

File: src/proxy-ui-api/frontend/src/components/layout/TabsBase.vue [hunk 1]
```diff
@@ -72,16 +73,9 @@ export default Vue.extend({
...
+    ...mapState(useUser, ['getAllowedTabs']),
     allowedTabs(): Tab[] {
-      return this.$store.getters.getAllowedTabs(mainTabs);
-    },
-  },
-  methods: {
-    logout(): void {
-      this.$store.dispatch('logout');
-      this.$router.replace({ name: RouteName.Login });
+      return this.getAllowedTabs(mainTabs);
```
[the remark this answers is not in the document]
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/components/layout/TabsBase.vue:78]: It was not actually used in this file. It was left behind from some reconstruction.

File: src/proxy-ui-api/frontend/src/store/modules/tokens.ts [hunk 2]
```diff
@@ -23,193 +23,166 @@
...
-    return api
-      .put(`/tokens/${encodePathParameter(id)}/logout`, {})
-      .then(() => {
-        // Update tokens
-        dispatch('fetchTokens');
-        dispatch('checkAlertStatus');
-      })
-      .catch((error) => {
-        throw error;
+      if (index === -1) {
```
[the remark this answers is not in the document]
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/store/modules/tokens.ts:145]: Mostly in the same order as they were in vuex so it's probably quite random order. Should be fixed now.

File: src/proxy-ui-api/frontend/src/store/modules/notifications.ts [hunk 3]
```diff
@@ -115,92 +101,91 @@ function createEmptyNotification(timeout: number): Notification {
...
-    return state.errorNotifications;
+  getters: {
+    getSuccessNotifications(state) {
+      return state.successNotifications;
+    },
+    getErrorNotifications(state) {
+      return state.errorNotifications;
+    },
+    getContinueInit(state) {
+      return state.continueInitialisation;
```
[the remark this answers is not in the document]
Comment (Riippi) [reply src/proxy-ui-api/frontend/src/store/modules/notifications.ts]: Removed

**Review by teemukin65 — APPROVED** [review]

**Pull request merged**